src/editor/src/routes.js

```diff
@@ -19,22 +14,27 @@ export default {
         workbox.routing.registerRoute(
             editRegex,
             async ({ url, event }) => {
+                const path = url.pathname.match(editRegex)[1];
                 const paths = splitURL(url.pathname);
 
                 let folder = file = redirectURL = "";
 
-                // If URL contains a file
-                if (isFile(paths[paths.length - 1])) {
-                    file = paths[paths.length - 1];
-                    paths.pop();
-                    folder = paths.join('/');
-                    redirectURL = `${url.origin}/editor?folder=${folder}&file=${file}`;
-                }
-                else {
-                    folder = paths.join('/');
-                    redirectURL = `${url.origin}/editor?folder=${folder}`;
-                }
+                fs.stat(path, (err, stats) => {
```

Comment:
Becuase `fs.stat()` is an async function, you have to wait on that callback before you run the code on line 38 below.  You need something like this:

```js
// Untested Code!!!!
fs.stat(path, (err, stats) => {
    if(err) {
        redirectURL = ...;
    } else {
        if(stats.isDirectory()) {
            ...
        } else {
            ...
        }
    }

    Response.redirect(redirectURL);
});
```